lxd/api_metrics.go

```diff
@@ -296,8 +278,40 @@ func metricsGet(d *Daemon, r *http.Request) response.Response {
 		metricsCache = map[string]metricsCacheEntry{}
 	}
 
+	// Add counter metrics for instances to the metric cache.
+	// We need to do create a distinct metric set for each project.
+	counterMetrics := make(map[string]*metrics.MetricSet, len(allProjectInstances))
+	for project, instanceCountMap := range allProjectInstances {
+		counterMetricSetPerProject := metrics.NewMetricSet(nil)
+		counterMetricSetPerProject.AddSamples(
+			metrics.Containers,
+			metrics.Sample{
+				Labels: map[string]string{"project": project},
+				Value:  float64(instanceCountMap[instancetype.Container]),
+			},
+		)
+		counterMetricSetPerProject.AddSamples(
+			metrics.VMs,
+			metrics.Sample{
+				Labels: map[string]string{"project": project},
+				Value:  float64(instanceCountMap[instancetype.VM]),
+			},
+		)
+
+		counterMetrics[project] = counterMetricSetPerProject
+	}
```

Comment:
After thinking some more, I wonder if we shouldn't go with a single metric but include a `type` label?

```
lxd_instances{project="default", type="container"} X
lxd_instances{project="default", type="vm"} Y
```

This would better fit with our other metrics that also have that `type` label:

```
lxd_procs_total{name="apt",project="default",type="container"} 13
```

Reply:
fixed

Reply:
Thanks and sorry for coming up with that so late :/